### PR TITLE
cocogitto: init as 4.0.1

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change
Tool for creating Conventional Commits

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/cocogitto/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    sha256 = "sha256-uSKzHo1lEBiXsi1rOKvfD2zVlkAUVZ5k0y8iiTXYE2A=";
+  };
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+  };
+
+  doCheck = false;
+
+  meta = with lib; {
+    description = "A set of cli tools for the conventional commit and semver specifications.";
```
Comment (legendofmiracles) [inline pkgs/development/tools/cocogitto/default.nix:18]:
```suggestion
    description = "A set of cli tools for the conventional commit and semver specifications";
```

File: pkgs/development/tools/cocogitto/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    homepage = "https://github.com/oknozor/cocogitto.git";
```
Comment (legendofmiracles) [inline pkgs/development/tools/cocogitto/default.nix:19]:
```suggestion
    homepage = "https://github.com/oknozor/cocogitto";
```

File: pkgs/development/tools/cocogitto/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+rustPlatform.buildRustPackage rec {
+  pname = "cocogitto";
+  version = "4.0.1";
+  src = fetchFromGitHub {
+    owner = "oknozor";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-uSKzHo1lEBiXsi1rOKvfD2zVlkAUVZ5k0y8iiTXYE2A=";
+  };
```
Comment (legendofmiracles) [inline pkgs/development/tools/cocogitto/default.nix:10]:
```suggestion
  };

```

File: pkgs/development/tools/cocogitto/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+rustPlatform.buildRustPackage rec {
+  pname = "cocogitto";
+  version = "4.0.1";
```
Comment (legendofmiracles) [inline pkgs/development/tools/cocogitto/default.nix:4]:
```suggestion
  version = "4.0.1";

```

File: pkgs/development/tools/cocogitto/default.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    owner = "oknozor";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-uSKzHo1lEBiXsi1rOKvfD2zVlkAUVZ5k0y8iiTXYE2A=";
+  };
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+  };
+
+  doCheck = false;
```
Comment (legendofmiracles) [inline pkgs/development/tools/cocogitto/default.nix:15]: Can you add a comment why the tests can't be run?

File: pkgs/development/tools/cocogitto/default.nix [hunk 6]
```diff
@@ -0,0 +1,23 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+rustPlatform.buildRustPackage rec {
```
Comment (legendofmiracles) [inline pkgs/development/tools/cocogitto/default.nix:2]:
```suggestion

rustPlatform.buildRustPackage rec {
```

File: pkgs/development/tools/cocogitto/default.nix (hunk 5 above)
Comment (TravisDavis-ops) [reply pkgs/development/tools/cocogitto/default.nix:18]: some test fall, I haven't looked in detail why.  just assumed it was due to nix and not the  package. considering the tool's test work as intended outside of build env

**Comment by TravisDavis-ops** [comment]
Hey what yawl using for formatting `nixpkg-fmt ` or `nixfmt` these are the only ones I know but they have different ideas about formatting

**Comment by legendofmiracles** [comment]
As the name suggests, `nixpkgs-fmt` is used in nixpkgs at the current time. But to what extent isn't really clear, RFC 101 tries to make it clear: https://github.com/NixOS/rfcs/pull/101

File: pkgs/development/tools/cocogitto/default.nix (hunk 5 above)
Comment (TravisDavis-ops) [reply pkgs/development/tools/cocogitto/default.nix:18]: so I have look more closely at the  error is seem to depend on some git setting being already set in to home environment of the build system,  Ill be updating this later today. sorry for the wait

Comment (legendofmiracles) [reply pkgs/development/tools/cocogitto/default.nix:21]: If you feel like it, it's very common for derivations to setup git a bit in preCheck, see [here](https://sourcegraph.com/search?q=context:global+git+config+repo:nixos/nixpkgs&patternType=literal) for examples.

Comment (TravisDavis-ops) [reply pkgs/development/tools/cocogitto/default.nix:21]: cool ill see what i can do to fix it up a bit

**Comment by legendofmiracles** [comment]
I made some adjustments and fixes in this patch: http://0x0.st/orou.txt.

Tried to fix the tests, but couldn't get it to work soon thereafter, so I just gave up :)

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 151131` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cocogitto</li>
  </ul>
</details>

**Review by legendofmiracles — APPROVED** [review]

**Comment by legendofmiracles** [comment]
I've gone ahead and applied my patch + fixed the build for darwin and merged your PR @TravisDavis-ops. Hope that's ok with you :)

**Comment by TravisDavis-ops** [comment]
👍 np, If there are more impovements I could make pls let me know

**Pull request merged**